src/IW/Effects/Download.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
```

Comment:
I wonder, does this extension is needed here?

Reply:
@chshersh I was testing something out and forgot to remove it. It's unnecessary now